File: perl/temperature_converter/temperature_converter_5.pl
#!/usr/bin/perl
use strict;
use warnings;

# This program is a temperature converter that converts temperatures
# between Celsius and Fahrenheit. It is designed to help programmers
# understand how to implement basic input/output operations, perform
# arithmetic calculations, and use conditional statements in Perl.

# Function to convert Celsius to Fahrenheit
sub celsius_to_fahrenheit {
    my ($celsius) = @_;
    my $fahrenheit = ($celsius * 9 / 5) + 32;
    return $fahrenheit;
}

# Function to convert Fahrenheit to Celsius
sub fahrenheit_to_celsius {
    my ($fahrenheit) = @_;
    my $celsius = ($fahrenheit - 32) * 5 / 9;
    return $celsius;
}

# Main program
print "Welcome to the Temperature Converter!\n";
print "Please choose an option:\n";
print "1. Convert Celsius to Fahrenheit\n";
print "2. Convert Fahrenheit to Celsius\n";
print "Enter your choice (1 or 2): ";
my $choice = <STDIN>;
chomp($choice);

if ($choice == 1) {
    print "Enter temperature in Celsius: ";
    my $celsius = <STDIN>;
    chomp($celsius);
    my $fahrenheit = celsius_to_fahrenheit($celsius);
    print "Temperature in Fahrenheit: $fahrenheit\n";
} elsif ($choice == 2) {
    print "Enter temperature in Fahrenheit: ";
    my $fahrenheit = <STDIN>;
    chomp($fahrenheit);
    my $celsius = fahrenheit_to_celsius($fahrenheit);
    print "Temperature in Celsius: $celsius\n";
} else {
    print "Invalid choice. Please run the program again and enter 1 or 2.\n";
}

# End of the program

File: vbnet/temperature_converter/temperature_converter_5.vb
' This program is a temperature converter written in Visual Basic .NET.
' It converts temperatures between Celsius and Fahrenheit.
' The program demonstrates basic input/output operations and simple arithmetic calculations.
' It is designed to help programmers understand the structure of a VB.NET program.

Module TemperatureConverter

    ' Main subroutine where the program execution begins
    Sub Main()
        ' Declare variables to hold the temperature values
        Dim celsius As Double
        Dim fahrenheit As Double

        ' Prompt the user to enter a temperature in Celsius
        Console.WriteLine("Enter temperature in Celsius:")
        ' Read the input from the user and convert it to a Double
        celsius = Convert.ToDouble(Console.ReadLine())

        ' Convert the Celsius temperature to Fahrenheit
        ' The formula for conversion is: F = (C * 9/5) + 32
        fahrenheit = (celsius * 9 / 5) + 32

        ' Display the converted temperature in Fahrenheit
        Console.WriteLine("Temperature in Fahrenheit: " & fahrenheit)

        ' Prompt the user to enter a temperature in Fahrenheit
        Console.WriteLine("Enter temperature in Fahrenheit:")
        ' Read the input from the user and convert it to a Double
        fahrenheit = Convert.ToDouble(Console.ReadLine())

        ' Convert the Fahrenheit temperature to Celsius
        ' The formula for conversion is: C = (F - 32) * 5/9
        celsius = (fahrenheit - 32) * 5 / 9

        ' Display the converted temperature in Celsius
        Console.WriteLine("Temperature in Celsius: " & celsius)

        ' Wait for the user to press a key before closing the program
        Console.WriteLine("Press any key to exit.")
        Console.ReadKey()
    End Sub

End Module

